sw/host/opentitanlib/src/proxy/handler.rs

```diff
@@ -0,0 +1,238 @@
+                        instance.write(data)?;
+                        Ok(Response::Uart(UartResponse::Write))
+                    }
+                }
+            }
+            Request::Spi { id, command } => {
+                let instance = self.transport.spi(id)?;
+                match command {
+                    SpiRequest::GetTransferMode => {
+                        let mode = instance.get_transfer_mode()?;
+                        Ok(Response::Spi(SpiResponse::GetTransferMode { mode }))
+                    }
+                    SpiRequest::SetTransferMode { mode } => {
+                        instance.set_transfer_mode(*mode)?;
+                        Ok(Response::Spi(SpiResponse::SetTransferMode))
+                    }
+                    SpiRequest::GetBitsPerWord => {
+                        let bits_per_word = instance.get_bits_per_word()?;
+                        Ok(Response::Spi(SpiResponse::GetBitsPerWord { bits_per_word }))
+                    }
+                    SpiRequest::SetBitsPerWord { bits_per_word } => {
+                        instance.set_bits_per_word(*bits_per_word)?;
+                        Ok(Response::Spi(SpiResponse::SetBitsPerWord))
+                    }
+                    SpiRequest::GetMaxSpeed => {
+                        let speed = instance.get_max_speed()?;
+                        Ok(Response::Spi(SpiResponse::GetMaxSpeed { speed }))
+                    }
+                    SpiRequest::SetMaxSpeed { value } => {
+                        instance.set_max_speed(*value)?;
+                        Ok(Response::Spi(SpiResponse::SetMaxSpeed))
+                    }
+                    SpiRequest::GetMaxTransferCount => {
+                        let number = instance.get_max_transfer_count()?;
+                        Ok(Response::Spi(SpiResponse::GetMaxTransferCount { number }))
+                    }
+                    SpiRequest::GetMaxChunkSize => {
+                        let size = instance.max_chunk_size()?;
+                        Ok(Response::Spi(SpiResponse::GetMaxChunkSize { size }))
+                    }
+                    SpiRequest::SetVoltage { voltage } => {
+                        instance.set_voltage(*voltage)?;
+                        Ok(Response::Spi(SpiResponse::SetVoltage))
+                    }
+                    SpiRequest::RunTransaction { transaction: reqs } => {
+                        // Construct proper response to each transfer in request.
+                        let mut resps: Vec<SpiTransferResponse> = reqs
+                            .iter()
+                            .map(|transfer| match transfer {
+                                SpiTransferRequest::Read { len } => SpiTransferResponse::Read {
+                                    data: vec![0; *len as usize],
+                                },
+                                SpiTransferRequest::Write { data: _ } => SpiTransferResponse::Write,
```

Comment:
You can also do `{ .. }` to ignore all fields. Applies throughout.

Reply:
Done